app/grandchallenge/cases/tasks.py

```diff
@@ -125,23 +129,43 @@ def build_images(  # noqa:C901
         The app_label of the linked object.
     linked_model_name:
         The model_name of the linked object.
+    linked_interface_slug:
+        The slug of the linked interface.
     """
 
     upload_session = lock_model_instance(
         pk=upload_session_pk,
         app_label=RawImageUploadSession._meta.app_label,
         model_name=RawImageUploadSession._meta.model_name,
     )
+    if linked_interface_slug:
+        ci = ComponentInterface.objects.get(slug=linked_interface_slug)
+    else:
+        ci = None
 
     if linked_object_pk:
-        linked_object = get_model_instance(
-            app_label=linked_app_label,
-            model_name=linked_model_name,
-            pk=linked_object_pk,
-        )
+        try:
+            linked_object = get_model_instance(
+                app_label=linked_app_label,
+                model_name=linked_model_name,
+                pk=linked_object_pk,
+            )
+        except ObjectDoesNotExist as e:
+            if linked_model_name in [
+                ArchiveItem._meta.model_name,
+                DisplaySet._meta.model_name,
+            ]:
+                # users can delete archive items and display sets before this task runs
```

Comment:
Why would we continue in this case? There's nothing to add the image to so why continue to import it?

Reply:
Good point! We shouldn't continue here.